.github/workflows/docker-build.yaml

```diff
@@ -27,29 +27,49 @@ jobs:
       matrix:
         image: ${{ fromJson(needs.list-images.outputs.images) }}
     steps:
+      - uses: actions/checkout@v3
       - name: Set Env
         run: |
-          echo "TOOL=$(basename `dirname "${{ matrix.image }}"` | awk '{print tolower($0)}')" >> $GITHUB_ENV
-          echo "SEMVER=$(basename "${{ matrix.image }}")" >> $GITHUB_ENV
-          echo "IMAGE_TAG=$(basename "${{ matrix.image }}")" >> $GITHUB_ENV
+          echo "TOOL=$(jq -r .name ${{ matrix.image }}/package.json)" >> $GITHUB_ENV
+          echo "SEMVER=$(jq -r .version ${{ matrix.image }}/package.json)" >> $GITHUB_ENV
           echo "GH_REF=${GITHUB_REF##*/}" >> $GITHUB_ENV
+      - name: Set Main Tag
+        if: github.ref == 'refs/heads/main'
+        run: echo "IMAGE_TAG=${{ env.SEMVER }}" >> $GITHUB_ENV
       - name: Set Branch Tag
         if: github.ref != 'refs/heads/main'
         run: echo "IMAGE_TAG=branch-${GITHUB_REF##*/}-${{ env.SEMVER }}" >> $GITHUB_ENV
-      - uses: actions/checkout@v3
       - name: Log in to the Container registry
         run: echo ${{ secrets.GITHUB_TOKEN }} | docker login ${{ env.REGISTRY }} --username ${{ github.actor }} --password-stdin
       - uses: webfactory/ssh-agent@v0.7.0
         with:
           ssh-private-key: ${{ secrets.STJUDE_SEQERR_READONLY_DEPLOY_KEY }}
       - uses: docker/setup-qemu-action@v2
       - uses: docker/setup-buildx-action@v2
+        id: build
       - uses: docker/build-push-action@v4
         with:
           context: ${{ matrix.image }}
           build-contexts: scripts=./scripts
-          push: true
+          push: false
+          load: true
           ssh: default
-          tags: ${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.TOOL }}:${{ env.IMAGE_TAG }}
+          tags: ${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.TOOL }}:local
           cache-from: type=registry,ref=${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.TOOL }}:buildcache
           cache-to: type=registry,ref=${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.TOOL }}:buildcache,mode=max
+      - name: Check digest
+        if: github.ref == 'refs/heads/main'
+        run: |
+          remote_digest=$(docker inspect ${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.TOOL }}:${{ env.IMAGE_TAG }} --format='{{index .RepoDigests 0}}')
+          local_digest=${{ steps.build.outputs.digest }}
+          if [ "$remote_digest" != "$local_digest" ]; then
+            echo "Digests do not match!"
+            echo "Remote: $remote_digest"
+            echo "Local: $local_digest"
+            echo "Should you increment the version number?"
+            exit 1
+          fi
```

Comment:
This has not run yet (because it requires being on `main`) so could use some extra scrutiny

Reply:
So right now `IMAGE_TAG` gets set dynamically. I think to do what you're suggesting here, we would need to have access to a `MAIN_IMAGE_TAG` and a `BRANCH_IMAGE_TAG`. Then `remote_digest` should get set to the digest of `MAIN_IMAGE_TAG`. If the digests differ, we error and complain?

Reply:
Pushed something up. Let's see how if it runs correctly!